src/map.cpp

```diff
@@ -863,9 +863,9 @@ int_fast32_t AStarNodes::getTileWalkCost(const Creature& creature, const Tile* t
 // Floor
 Floor::~Floor()
 {
-	for (uint32_t i = 0; i < FLOOR_SIZE; ++i) {
-		for (uint32_t j = 0; j < FLOOR_SIZE; ++j) {
-			delete tiles[i][j];
+	for (auto& tile : tiles) {
+		for (auto& j : tile) {
```

Comment:
j is already a pointer to a tile here, so we don't need a reference to it since copying pointers is cheaper than accessing it by another indirection. You could also change it to explicitly say Tile* instead of auto& so that it's easier to understand that when looking at this code.